bittide/src/Bittide/DoubleBufferedRam.hs

```diff
@@ -564,3 +564,57 @@ updateAddrs rdAddr (Just (i, a)) bufSelect =
 
 updateAddrs rdAddr Nothing bufSelect =
   (mul2Index rdAddr bufSelect, Nothing)
+
+{-# NOINLINE fifo #-}
+
+-- | Simple First in, First out buffer the contains a backpressure mechanism.
+fifo ::
+  forall dom fifoDepth a .
```

Comment:
```suggestion
  forall depth a dom .
```

I'm assuming `depth` is the type argument you want to type apply most often, so it should go first. Also renamed to `depth` (from `fifoDepth`), because this function is already called `fifo`.